torch/_subclasses/meta_utils.py

```diff
@@ -102,10 +102,13 @@ def go(m1, m2):
         assert_eq(m1.grad is not None, safe_grad(m2) is not None)
         if m1.grad is not None:
             go(m1.grad, safe_grad(m2))
+        assert_eq(m1.layout, m2.layout)
         if m1.is_sparse:
             assert_eq(m1.dense_dim, m2.dense_dim())
             assert_eq(m1.sparse_dim, m2.sparse_dim())
             assert_eq(m1.is_coalesced, m2.is_coalesced())
+        elif is_sparse_compressed(m1):
+            assert_eq(m1.sparse_dim, m2.sparse_dim())
```

Comment:
Sparse compressed tensors have also dense dimensions:
```suggestion
            assert_eq(m1.sparse_dim, m2.sparse_dim())
            assert_eq(m1.dense_dim, m2.dense_dim())
```

Reply:
added